src/AgentBuilder.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+
+namespace AG\ElasticApmLaravel;
+
+
+use AG\ElasticApmLaravel\Collectors\EventDataCollector;
+use AG\ElasticApmLaravel\Collectors\RequestStartTime;
+use Illuminate\Support\Collection;
+use Nipwaayoni\AgentBuilder as NipwaayoniAgentBuilder;
+use Nipwaayoni\ApmAgent;
+use Nipwaayoni\Config;
+use Nipwaayoni\Contexts\ContextCollection;
+use Nipwaayoni\Events\EventFactoryInterface;
+use Nipwaayoni\Middleware\Connector;
+use Nipwaayoni\Stores\TransactionsStore;
+
+class AgentBuilder extends NipwaayoniAgentBuilder
+{
+    /** @var RequestStartTime */
+    private $startTime;
+
+    /** @var Collection */
+    private $collectors;
+
+    public function withRequestStartTime(RequestStartTime $startTime): self
```

Comment:
I prefer the `with` nomenclature because of the builder pattern behavior. We're providing details to the builder for it to use when creating an `Agent`, not really setting something on the builder itself. The `with` methods also return an instance of the builder object to enable fluent chaining, which is not a normal setter behavior.

The Agent package is also not coupled to Laravel, so I do not follow Laravel specific conventions. It's used in a number of non-Laravel PHP projects, so the conventions should be framework agnostic. That will hidden somewhat by your service provider and facade.